File: Api/QLyLapTop/QLyLapTop/MyModels/SanPham.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace QLyLapTop.MyModels;

[Table("SanPham")]
public partial class SanPham
{
    [Key]
    public int MaSanPham { get; set; }

    [StringLength(255)]
    public string TenSanPham { get; set; } = null!;

    public int? MaThuongHieu { get; set; }

    [StringLength(255)]
    public string? XuatXu { get; set; }

    [Column(TypeName = "decimal(10, 2)")]
    public decimal? GiaNhap { get; set; }

    [Column(TypeName = "decimal(10, 2)")]
    public decimal? GiaXuat { get; set; }

    public int? SoLuongTonKho { get; set; }

    [StringLength(255)]
    public string? KhuVucKho { get; set; }

    [StringLength(255)]
    public string? HinhAnh { get; set; }

    [StringLength(255)]
    public string? Ram { get; set; }

    [Column("CPU")]
    [StringLength(255)]
    public string? Cpu { get; set; }

    public string? MoTa { get; set; }

    public int? TrangThai { get; set; }

   
}

File: Flutter/lib/model/SanPham.dart
class Sanpham {
  final int maSanPham;
  final String tenSanPham;
  final int maThuongHieu;
  final String xuatXu;
  final double giaXuat;
  final int soLuongTonKho;
  final String hinhAnh;
  final String Ram;
  final String CPU;
  final String moTa;
  final int trangThai;

  Sanpham(
      {required this.maSanPham,
      required this.tenSanPham,
      required this.maThuongHieu,
      required this.xuatXu,
      required this.giaXuat,
      required this.soLuongTonKho,
      required this.hinhAnh,
      required this.Ram,
      required this.CPU,
      required this.moTa,
      required this.trangThai});

  factory Sanpham.fromJson(Map<String, dynamic> json) {
    return Sanpham(
        maSanPham: json['maSanPham'],
        tenSanPham: json['tenSanPham'],
        maThuongHieu: json['maThuongHieu'],
        xuatXu: json['xuatXu'],
        giaXuat: json['giaXuat'],
        soLuongTonKho: json['soLuongTonKho'],
        hinhAnh: json['hinhAnh'],
        Ram: json['ram'],
        CPU: json['cpu'],
        moTa: json['moTa'],
        trangThai: json['trangThai']);
  }
}
